README.md
# Anarcho Notepad

**Free WordPress Theme**

Inspired by the idea of anarchy, we designed this free WordPress theme for your personal blogs and diaries. This theme is perfect for writers and artists. "Anarcho Notepad" can be easily customized and already translated to different languages. It utilizes latest HTML-5, CSS-3, PHP and WordPress native functions for creating the awesomeness that looks good on every browser. We are constantly adding new features to this theme to allow you to personalize it to your own needs. If you want a new feature or just want to be able to change something just ask us and we would be happy to add it. We are glad to present you the Most Anarchist WP-Theme Of All! Enjoy!

[![screenshot](https://github.com/ArthurGareginyan/anarcho-notepad/blob/master/screenshot.png)]()

>This theme is used simultaneously by more than 3.000 people worldwide!


## Features

* Hand-coded, no WYSIWYG stuff
* HTML5 Inside
* CSS3 Inside
* Easy to set up – Theme Customizer
* Customize using “Google Web Fonts”
* Built-in iconic font “Font Awesome”
* Widget ready (in sidebar and footer)
* Built-in Pagination
* Built-in Breadcrumbs
* Template page “HTML map of the website”
* Template page “Without date”
* Block “About Author” below posts
* Custom text of copyright below posts
* Custom text of copyright in footer
* Custom header image
* Custom background
* Custom menu
* Ready for translation (Translated to 12 languages)
* Published to the repository WordPess.org

**Got more ideas? Tell me!**


## Contributing

Thank you for considering contributing to the theme "Anarcho Notepad"! The contribution guide can be found in the [CONTRIBUTING.md](https://github.com/ArthurGareginyan/anarcho-notepad/blob/master/CONTRIBUTING.md).


## Security Vulnerabilities

If you discover a security vulnerability within this theme, please send an email to Arthur Gareginyan at arthurgareginyan@gmail.com. All security vulnerabilities will be promptly addressed.


## Copyrights for Resources used in this theme

1. The "Font Awesome" iconic font are under the MIT License. http://fontawesome.io
2. The "Forum" font are under the SIL Open Font License v1.1.
3. The html5.js file present in the "/js" folder are under the MIT/GPL2 License. https://github.com/aFarkas/html5shiv
4. The follow.png file in the "/images" folder are under the Creative Commons License. By Colaja. https://www.iconfinder.com/icons/87493/rss_icon#size=128
5. The notepad-middle.png, yellow-sticker-middle, yellow-sticker-bottom, white-sticker-top, white-sticker-middle, white-sticker-bottom images are from the theme ("Notepad Chaos v2.0")[http://www.notepadchaos.com] by (Evan Eckard)[http://www.evaneckard.com].
6. The anarchy-symbol-big.png and anarchy-symbol.png images is under public domain: http://www.layoutsparks.com/1/213916/anarchy-perfect-black-A-31000.html
7. The Default Background Image (images/background.png) is under public domain: http://wallbox.ru/wallpaper/9422/#.UnLDFNQhCRs

Everything else used in this theme has been created by me, especially for "Anarcho Notepad" theme.


## License

All PHP code, and PDF/POT/PO/MO files is released under the [GNU General Public License version 3 (GPLv3)](http://www.gnu.org/licenses/gpl-3.0.html).

All HTML/CSS/JAVASCRIPT code, and PNG files is released under the restrictive proprietary license.


## Author

Arthur Gareginyan

* Email: arthurgareginyan@gmail.com

* GitHub: [https://github.com/ArthurGareginyan/](https://github.com/ArthurGareginyan/)

* Website: [http://www.arthurgareginyan.com](http://www.arthurgareginyan.com)

* Donation: [http://www.arthurgareginyan.com/donate.html](http://www.arthurgareginyan.com/donate.html)


## Links

* [Dedicated Theme Page on SpaceXChimp.org](https://www.spacexchimp.com/themes/anarcho-notepad.html)
* [Dedicated Theme Page on WordPress.org](https://wordpress.org/themes/anarcho-notepad/)


---
#### Technologies:

[![HTML5](http://resources.spacexchimp.com/images/logos/HTML5.png)]()
[![CSS3](http://resources.spacexchimp.com/images/logos/CSS3.png)]()
[![PHP](http://resources.spacexchimp.com/images/logos/PHP.png)]()
[![JS](http://resources.spacexchimp.com/images/logos/JavaScript.png)]()
[![JQuery](http://resources.spacexchimp.com/images/logos/jQuery.png)]()
[![WordPress](http://resources.spacexchimp.com/images/logos/WordPress.png)](https://wordpress.org)
[![License](http://resources.spacexchimp.com/images/logos/GPLv3.png)](http://www.gnu.org/licenses/gpl-3.0.html)


CHANGELOG.txt
== Changelog ==

=
* Maintenance: Ensure compatibility with upcoming WordPress 5.8.

= 2.48 (2021.01.01) =
* Maintenance: The copyright date updated to support the 2021 year.

= 2.47 (2020.12.21) =
* Maintenance: Ensure compatibility with WordPress 5.6.
* Improvement: The "click-to-follow.me" URL for the "RSS feed" ribbon tag has been replaced with the new "www.follow.it/now" and switched to secure HTTPS protocol.
* Fix: The "RSS feed" ribbon tag is not showing up on the front end.
* Fix: Custom scripts inserted into the "Scripts" option text fields are added to the wrong locations. The "Scripts at the end of the posts" and "Scripts after posts" are mixed up.
* Fix: Custom scripts inserted into the "Scripts" option text fields displayed on the front end as raw code instead of being rendered by web browser. Options: "Scripts before posts", "Scripts at the beginning of the posts", "Scripts at the end of the posts" and "Scripts after posts".
* Fix: REQUIRED: Found echo get_template_directory_uri in sidebar.php. get_template_directory_uri() must be escaped when output as part of a link or image source.
* Fix: REQUIRED: Found echo get_template_directory_uri in inc/upgrade_notice.php. get_template_directory_uri() must be escaped when output as part of a link or image source.
* Fix: WARNING: Found echo get_theme_mod in functions.php. Theme options must be escaped.
* Fix: WARNING: Found echo get_theme_mod in single.php. Theme options must be escaped.
* Fix: The image in the upgrade notice is displayed too large for a moment.
* New feature: The "Copyright in footer" option now allows HTML tags to be passed. You can write a pretty much any HTML tags in the option's textfield (Appearance → Customize → Meta → "Copyright in footer") and it will be rendered before being showed up on the front end.
* New feature: The "Copyright after post" option now allows HTML tags to be passed. You can write a pretty much any HTML tags in the option's textfield (Appearance → Customize → Meta → "Copyright after post") and it will be rendered before being showed up on the front end.
* New feature: The "About widget - "What is this place?"" option now allows HTML tags to be passed. You can write a pretty much any HTML tags in the option's textfield (Appearance → Customize → Meta → "About widget - "What is this place?"") and it will be rendered before being showed up on the front end.
* Maintenance: PHP and HTML code has been better formatted.

= 2.46 (2020.08.28) =
* New feature: Added the 'wp_body_open()' function introduced in WordPress 5.2 to support the new function and action.
* Maintenance: To support WordPress versions prior to 5.2.0, added a function that registers the 'wp_body_open()' function if the site is using an older version of WordPress that does not have the 'wp_body_open()' function.
* Maintenance: Checking for the '_wp_render_title_tag' function and defining the 'spacexchimp_theme_render_title' function have been removed due to uselessness after discontinuing of support of WordPress versions prior to 4.1.

= 2.45 (2020.08.10) =
* Maintenance: Ensure compatibility with upcoming WordPress 5.5.
* Maintenance: Content of the CSS files has been beautified.
* Maintenance: Content of the CSS files has been sorted in SMACSS order (from most important, flow affecting properties, to least important properties).
* Maintenance: Content of the JS files has been beautified.

= 2.44 (2020.03.30) =
* Maintenance: Ensure compatibility with upcoming WordPress 5.4.
* Maintenance: Minimum WordPress version requirement is set to 4.9. Support for WordPress 4.8 and below has been discontinued.
* Maintenance: Minimum PHP version requirement is set to 5.6. Support for PHP 5.5 and below has been discontinued.
* Maintenance: Support of legacy web browsers (Internet Explorer 6-9, Safari 4.x, iPhone 3.x, Firefox 3.x) has been discontinued.
* Maintenance: The readme.txt content is corrected.
* Maintenance: Applied vendor prefixes in CSS rules using values from [Can I Use](https://caniuse.com/), the data based on current browser popularity and property support.

= 2.43 (2020.01.13) =
* Fix: Non-printable characters were found in the inc/theme_info.php file on line 110.
* Fix: Possible hard-coded links were found in the file inc/theme_info.php on line 84 and 117.
* Maintenance: The copyright date updated to support the 2020 year.

= 2.42 (2019.12.22) =
* New: The translation into Portuguese (Brazil) has been added. (Thanks to Fabio Correia Pimenta)

= 2.41 (2019.11.28) =
* New: The translation into Italian has been added. (Thanks to Demetrio Fiumanò)
* Improvement: The translation into Ukrainian has been updated.
* Improvement: The translation into Russian has been updated.
* Fix: Not all text strings were prepared for translation. (Thanks to Demetrio Fiumanò)

= 2.40 (2019.11.11) =
* Maintenance: Compatibility with upcoming WordPress 5.3.
* Framework update: The contents of the "humans.txt" file has been updated.
* Framework update: The contents of the "languages/Readme.txt" file has been updated to provide a more detailed explanation of the theme translation process to the user.
* Fix: The theme translation files de_DE and ru_RU have been corrected.

= 2.39 (2019.01.07) =
* The copyright date updated.
* The variables prefix replaced from "anarcho_" to "spacexchimp_theme_".
* The functions prefix replaced from "anarcho_" to "spacexchimp_theme_".
* The "spacexchimp_t001_define_constants" function renamed to "spacexchimp_theme_define_constants".
* All translation files are updated.

= 2.38 (2018.11.16) =
* Improvement: Code in PHP and CSS files is better formatted using the best coding standards.
* Improvement: The breadcrumbs on the Search page is redesigned.
* The "Font Awesome" icon in the "Help" button, which used in the top bar, is replaced with a new one from the "Dashicons".
* Fix: The closing tag "</h1>" did not match the opening tag "<h2>" in the 492 line of the file "functions.php".

sidebar.php
<?php
/**
 * The sidebar containing the secondary widget area, displays on posts and pages.
 *
 * @package     Anarcho Notepad
 * @since       2.47
 * @author      Space X-Chimp
 * @copyright   Copyright (c) 2013-2021, Space X-Chimp
 * @link        https://www.spacexchimp.com/themes/anarcho-notepad.html
 * @license     http://www.gnu.org/licenses/gpl-3.0.html
 */
?>

<aside id="sidebar" role="complementary">

    <!--FOLLOW-->
    <?php if ( get_theme_mod('disable_rss_icon') !== '1' ) { ?>
    <div class="follow">
        <a href="https://www.follow.it/now" target="_blank" rel="nofollow">
            <img src="<?php echo esc_url( get_template_directory_uri() ); ?>/images/follow.png" alt="RSS" />
        </a>
    </div>
    <?php } ?>
    <!--END-FOLLOW-->

    <!--YELLOW STICKER-->
    <nav class="yellow-sticker-middle">
        <?php wp_nav_menu( array( 'theme_location' => 'primary' ) ); ?>
    </nav>
    <div class="yellow-sticker-bottom"></div>
    <!--END-YELLOW STICKER-->

    <?php dynamic_sidebar( 'sidebar-1' ); ?>

    <!--ABOUT (WIDGET)-->
    <?php if ( get_theme_mod('disable_about_box') !== '1' ) { ?>
    <div class="about-widget">
        <p>
            <?php
                $about_text = __( 'Paste here small text about you and/or about your site.', 'anarcho-notepad' );
                echo wp_kses_post( get_theme_mod( 'about_box', $about_text ) );
            ?>
        </p>
    </div>
    <?php } ?>
    <!--END-ABOUT (WIDGET)-->

    <?php dynamic_sidebar( 'sidebar-2' ); ?>

    <!--LINKS (WIDGET)-->
    <?php if ( get_theme_mod('disable_links_box') !== '1' ) { ?>
    <div class="links-widget">
        <p>
            <ul>
                <?php wp_list_bookmarks('categorize=0&title_li=&orderby=rating'); ?>
            </ul>
        </p>
    </div>
    <?php } ?>
    <!--END-LINKS (WIDGET)-->

    <?php dynamic_sidebar( 'sidebar-3' ); ?>

    <!--WHITE STICKER-->
    <?php if ( get_theme_mod('disable_recent_sticker') !== '1' ) { ?>
    <div class="white-sticker-top"></div>
    <nav class="white-sticker-middle">
        <?php $num_recent_post = get_theme_mod('num_recent_post'); ?>
        <?php $query = new WP_Query( array( 'posts_per_page' => $num_recent_post ) ); ?>
        <?php while ( $query->have_posts() ) : $query->the_post(); ?>
        <ul>
            <li>
                <a href="<?php the_permalink(); ?>">
                    <i class="fa fa-asterisk"></i>
                    <?php the_title(); ?>
                </a>
            </li>
        </ul>
        <?php endwhile; ?>
    </nav>
    <div class="white-sticker-bottom"></div>
    <?php } ?>
    <!--END-WHITE STICKER-->

    <?php dynamic_sidebar( 'sidebar-4' ); ?>

</aside>

<br clear="all">
